### PR TITLE
feat: implementation of int-list

### PR DESCRIPTION
**Pull request by ealanisg**

### CODE REVIEW

File: src/param/int-list.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+const assert = require('assert');
+const List = require('./list');
+
+class IntList extends List {
```
Comment (krackerman) [inline src/param/int-list.js]: You should extend `NumberList` here not `List`. `NumberList` already has this validation in it. We don't want to duplicate the code here :)

File: src/param/number-list.js [hunk 2]
```diff
@@ -6,10 +6,10 @@ class NumberList extends List {
     super(name, position, opts);
     this.items = { type: 'number' };
 
-    assert(opts.minItemValue === undefined || Number.isInteger(opts.minItemValue));
+    assert(opts.minItemValue === undefined || (typeof opts.minItemValue === 'number'));
```
Comment (krackerman) [inline src/param/number-list.js]: I don't think you need the extra parenthesis here

File: test/param/int-list.spec.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
+const expect = require('chai').expect;
```
Comment (krackerman) [inline test/param/int-list.spec.js]: Much of this testing is already done in `NumberList`. So after you extend from `NumberList`, you will be able to remove a lot of these tests.

File: test/param/int-list.spec.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    let bodyParamInValidMin;
+    before(() => {
+      queryParamInValidMin = api.IntList('list', 'query', { minItemValue: 1 });
+      bodyParamInValidMin = api.IntList('list', 'json', { minItemValue: 1 });
+    });
+
+    it('Testing invalid min-value of list item in query param', () => {
+      expect(() => queryParamInValidMin.get({
+        queryStringParameters: {
+          list: '[0, 1.2]'
```
Comment (krackerman) [inline test/param/int-list.spec.js]:
```suggestion
          list: '[0, 1]'
```

Comment (ealanisg) [reply test/param/int-list.spec.js]: I'm going to let it as `list: '[0, 2]'` for consistency with JSON param test that is using same params

**Pull request merged**